dev-db/mariadb/mariadb-11.7.2.recipe

```diff
@@ -0,0 +1,160 @@
+SECONDARY_ARCHITECTURES="!x86"
+
+POST_INSTALL_SCRIPTS="
+	$relativePostInstallDir/setup-mariadb.sh
+	"
+
+PROVIDES="
+	mariadb = $portVersion
+	cmd:mariadb$secondaryArchSuffix
+	cmd:mariadbd$secondaryArchSuffix
+	lib:libmariadb$secondaryArchSuffix
+	"
+REQUIRES="
+	haiku
+	lib:libboost_system$secondaryArchSuffix
+#	lib:libcurl$secondaryArchSuffix
+#	lib:libncurses$secondaryArchSuffix
+	lib:libncursesw$secondaryArchSuffix
+	lib:libpcre2_8$secondaryArchSuffix
+	lib:libsnappy$secondaryArchSuffix
+	lib:libssl$secondaryArchSuffix
+	lib:libz$secondaryArchSuffix
+	"
+
+PROVIDES_devel="
+	mariadb${secondaryArchSuffix}_devel = $portVersion
+	devel:libmariadb$secondaryArchSuffix
+	devel:libmariadbclient$secondaryArchSuffix
+	devel:libmysqlclient$secondaryArchSuffix
+	devel:libmysqlclient_r$secondaryArchSuffix
+	devel:libmysqlservices$secondaryArchSuffix
+	"
+REQUIRES_devel="
+	mariadb$secondaryArchSuffix == $portVersion base
+	"
+
+BUILD_REQUIRES="
+	haiku${secondaryArchSuffix}_devel
+	devel:libboost_system$secondaryArchSuffix >= 1.83.0
+#	devel:libcurl$secondaryArchSuffix
+	devel:libmsgpackc$secondaryArchSuffix
+	devel:libncurses$secondaryArchSuffix
+	devel:libpcre2_8$secondaryArchSuffix
+	devel:libsnappy$secondaryArchSuffix
+	devel:libssl$secondaryArchSuffix
+	devel:libz$secondaryArchSuffix
+	devel:libzmq$secondaryArchSuffix
+	"
+BUILD_PREREQUIRES="
+	cmd:bison
+	cmd:cmake
+	cmd:g++$secondaryArchSuffix
+	cmd:make
+	cmd:ninja
+	cmd:pkg_config$secondaryArchSuffix
+	cmd:python3
+	cmd:which
+	"
+
+BUILD()
+{
+	# required sources
+	cp -r $sourceDir2/mariadb-connector-c-3.4.4/* $sourceDir/libmariadb/
+	cp -r $sourceDir3/wsrep-lib-70cd967f5e249b53d6cce90e2e4198641c564381/* $sourceDir/wsrep-lib
+	cp -r $sourceDir4/wsrep-API-master/* $sourceDir/wsrep-lib/wsrep-API/v26
+
+	# additional sources for libmarias3 / columnstore
+	cp -r $sourceDir5/libmarias3-0d5babbe46f17147ed51efd1f05a0001017a2aad/* $sourceDir/storage/maria/libmarias3
+	cp -r $sourceDir6/mariadb-columnstore-engine-9763b126517c8efb716e767fd5ba4eb2b5b405fc/* $sourceDir/storage/columnstore/columnstore
+
+	# mariadbd binary is with debug info (even if release mode is indicated)
+	cmake -B build -G Ninja \
+		-DCMAKE_BUILD_TYPE=Release \
+		-DWITHOUT_AIO=ON \
+		-DPLUGIN_TOKUDB=NO \
+		-DWITH_SYSTEMD=no \
+		-DPLUGIN_CONNECT=NO \
+		-DWITH_SSL=system \
+		-DCMAKE_EXE_LINKER_FLAGS="-lnetwork" \
+		-DCMAKE_SHARED_LINKER_FLAGS="-lnetwork" \
+		-DWITHOUT_ROCKSDB=ON \
+		-DWITH_UNIT_TESTS=NO \
+		-DCMAKE_INSTALL_PREFIX=$appsDir/MariaDB
+
+	cmake --build build $jobArgs
+}
+
+
+INSTALL()
+{
+	# post install script installation
+	mkdir -p $postInstallDir
+	cp -f $portDir/additional-files/setup-mariadb.sh $postInstallDir
+	chmod a+rx "$postInstallDir/setup-mariadb.sh"
+
+	# install mariadb
+	cmake --install build --prefix $appsDir/MariaDB
+
+	# mariadb data
+	mkdir -p /boot/home/config/var/mariadb/data
+
+	# mariadb settings : put a default config file
+	mkdir -p $settingsDir/mariadb
+
```

Comment:
Creating directories in a recipe is a nogo, should be done in the post-install script, I guess with the changes mentioned by @jmairboeck
On the matter for settings dir, I guess you could use the internal cmake value to pass the directory? (like you can see on the one I fiddled with)
```
			-DINSTALL_SYSCONFDIR=/etc \
			-DINSTALL_SYSCONF2DIR=/etc/my.cnf.d \
```

Reply:
Ok I mimick the nginx recipe to have $dataDir/user_launch/maria-db to launch the server during the system restart. It means I can add this part (directory create + user_launch addition) in the post-install script ?

Reply:
I'm not even sure if it's a good idea to add user/password without user interference, I'll let the masters decide on that :)